pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -212,6 +221,19 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
               dictionaryCreator.getNumBytesPerEntry());
           throw e;
         }
+
+        // A Bloom Filter can only be applied to dictionary-encoded columns TODO(saurabh) check on this
```

Comment:
For settings that cannot be applied to certain encoding, we should inform the users (maybe fail the validation). We may do it in a separate PR.